### ISSUE TITLE
Progress update intervals for GUI-managed backups

### ISSUE THREAD

**Issue by icgood** [restic/restic#1199]
## Output of `restic version`

```
restic 0.7.1
compiled with go1.8.3 on linux/arm
```
and from 7ed0f61f3fd5ef2df551a7559d874e7d087f6853
```
restic compiled manually
compiled with go1.9 on linux/arm
```

## How did you run restic exactly?

```bash
export RESTIC_REPOSITORY=s3:http://minio:9000/backup
export RESTIC_PASSWORD=
export AWS_ACCESS_KEY_ID=
export AWS_SECRET_ACCESS_KEY=

restic backup -x $HOME | cat
```

## What backend/server/service did you use?
Minio on local network

## Expected behavior
I am working on a simple GUI menubar app for backing up using restic, which would render the progress of a current backup. Obviously there is no terminal, so using `kill -USR1 <pid>` works for progress updates. However, doing a full system backup (as root) requires that I also be able to run `kill` as root, and there is no secure way to do that for the duration of the backup.

I would expect a way to force interval'ed progress updates for the purposes of programmatic backup management.

## Actual behavior
No progress updates are output, which appears to be by design as of 5247de552ac847b451f0450fa06658326e14ed8c.

## Proposed fix
Keep the default behavior for non-terminals of no output, but add an environment variable to force a progress update interval. I will affix a proposed patch as a comment, and can PR as necessary.

**Comment by icgood** [comment]
## Proposed patch

```diff
diff --git a/internal/restic/progress.go b/internal/restic/progress.go
index e5d608f0..579a2bfd 100644
--- a/internal/restic/progress.go
+++ b/internal/restic/progress.go
@@ -3,6 +3,7 @@ package restic
 import (
 	"fmt"
 	"os"
+	"strconv"
 	"sync"
 	"time"
 
@@ -52,7 +53,7 @@ type ProgressFunc func(s Stat, runtime time.Duration, ticker bool)
 // OnDone is called when Done() is called. Both functions are called
 // synchronously and can use shared state.
 func NewProgress() *Progress {
-	var d time.Duration
+	d := getProgressIntervalEnv()
 	if isTerminal {
 		d = time.Second
 	}
@@ -217,3 +218,13 @@ func (s Stat) String() string {
 	return fmt.Sprintf("Stat(%d files, %d dirs, %v trees, %v blobs, %d errors, %v)",
 		s.Files, s.Dirs, s.Trees, s.Blobs, s.Errors, str)
 }
+
+func getProgressIntervalEnv() time.Duration {
+	if e := os.Getenv("RESTIC_PROGRESS_INTERVAL"); e != "" {
+		i, err := strconv.ParseInt(e, 0, 0)
+		if err == nil {
+			return time.Duration(i) * time.Second
+		}
+	}
+	return 0
+}
```

**Comment by fd0** [comment]
Thanks for the suggestion, this will dealt with eventually (but not right now). :)

**Comment by bherila** [comment]
@icgood is your restic menubar app open source? Interested in it (and possibly helping out too!) 😄

**Comment by icgood** [comment]
@bherila It's for macOS specifically, written as a Hammerspoon plugin:

https://github.com/icgood/Restic.spoon

It works nicely for me so far. It will read and render any progress output from the restic process, but I have not figured out a good way to trigger that output (hence this issue). From a terminal, you can try sending a signal to the process manually to see it (e.g. `sudo killall -USR1 restic`). Because I start the restic process as root with a password prompt, there's no good way I could think of to trigger regular signals to the process.

**Comment by remss** [comment]
Why using an environment variable instead of a CLI option ?

**Comment by odinho** [comment]
I'm running restic in systemd with `--verbose`, but there's no progress indication other than "uploaded intermedia index <short sha>". This is still expected without this bug having been solved, right, nothing I missed? (tried killing with `USR1` too, but no output) :)

**Comment by Elinvention** [comment]
@odinho I have the same problem: I started the backup with systemd but I have no way to get a percentage or ETA. I probably should have started the first backup by hand.

**Comment by rawtaz** [comment]
@Elinvention If you are still on a large initial backup, you can just cancel it and start it again manually if you want. Besides the initial scanning for files, restic will only upload what hasn't been uploaded before, so it's perfectly fine to resume/restart.

**Comment by MichaelEischer** [comment]
This has been fixed by #3199 which is part of restic 0.12.0.
